### PR TITLE
Check that requests exception has a response (#314)

### PR DESCRIPTION
**Pull request by EricRahm**
It's possible for a RequestException to not have a response,
for example if a connection times out a Response object will
not have been created.

@whimboo Do you mind taking a look?

### CODE REVIEW

**Comment by EricRahm** [comment]
@parkouss Much simpler, this look good to you?

**Comment by parkouss** [comment]
Yup, looks good to me! @whimboo should have the last word though. :)

File: mozdownload/scraper.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ def _retry_check_404(self, func,
                                  requests.exceptions.RequestException))
         try:
             self._retry(func, **retry_kwargs)
-        except requests.exceptions.RequestException as exc:
+        except requests.exceptions.HTTPError as exc:
```
Comment (whimboo) [inline mozdownload/scraper.py]: This change looks fine to me. Lets see which ambiguous exception has been raised by requests in those rare occurrences.

**Comment by whimboo** [comment]
Merged PR as https://github.com/mozilla/mozdownload/commit/69e067ab6188701c9166233083106f276c1459fd.

**Pull request closed without merging**